### PR TITLE
Fat gene now makes you fat

### LINKED ISSUE #27660: Toggling the FAT gene doesnt actually make someone fat
(WEB REPORT BY: chazach REMOTE: 206.221.180.138:7777)
# Revision
34b7e27544426513a953120e983ab3f27913f9c5

# Description
says on tin
# Steps to Reproduce
Toggle FAT gene on admin panel
# What you Expected
epic pwn dumb ablinos
# What Actually Happened
nothing

### PR DESCRIPTION
**Pull request by Stressedphysicist**
Closes #27660 

:cl:
 * tweak: Obesity and Fatness are now the same thing, being fat makes it harder to unfat
 * bugfix: The fat gene now makes you fat, removing it doesn't make you unfat

### CODE REVIEW

File: code/game/dna/genes/goon_disabilities.dm [hunk 1]
```diff
@@ -72,6 +72,19 @@
 
 	return 1
 
+/datum/dna/gene/disability/fat/activate(var/mob/M)
+	if(M.overeatduration < 100)
+		M.overeatduration += 600 // 500 is minimum fat threshold.
+	else
```
Comment (DamianX) [inline code/game/dna/genes/goon_disabilities.dm]: This entire else clause does nothing, remove it.

File: code/game/dna/genes/goon_disabilities.dm [hunk 2]
```diff
@@ -72,6 +72,19 @@
...
+		return
+
+/datum/dna/gene/disability/fat/deactivate(var/mob/M)
+	if(M.overeatduration > 100)
+		M.overeatduration = 0
+		M.nutrition = max(M.nutrition - 200, 0)
+	else
```
Comment (DamianX) [inline code/game/dna/genes/goon_disabilities.dm]: This entire else clause does nothing, remove it.

**Review by DamianX — COMMENTED** [review]
This could be okay, but I wonder if instead of changing `overeatduration` itself, it could be better to make the code in `Life()` that reads `overeatduration` check for the fat gene?

**Comment by Stressedphysicist** [comment]
> 
> 
> This could be okay, but I wonder if instead of changing `overeatduration` itself, it could be better to make the code in `Life()` that reads `overeatduration` check for the fat gene?

The thing is that `M_Fat` (the mutation that makes you fat) depends on the level of `overeatduration` and most of the code I have seen that deals with making the user fat works by setting `overeatduration = 600`.

I changed the Proc of the gene so the person stays fat until the gene is removed, but this might make M_Obesity useless

**Comment by Stressedphysicist** [comment]
you press fat on admin panel, that person now fat
you activate fat block on thin man and he is now fat
simple as

**Comment by Zth--** [comment]
I'd like to see M_OBESITY merged into M_FAT. My reasoning is that becoming fat in-game is different than starting as a fat person and I don't see a justification for that inconsistency. It also makes sense that once you become fat is harder to become thin again

**Review by Zth-- — CHANGES_REQUESTED** [review]
remove M_OBESITY from setup.dm

**Review by Zth-- — APPROVED** [review]

**Pull request merged**